pkg/patterns/declarative/reconciler.go

```diff
@@ -122,6 +122,22 @@ func (r *Reconciler) reconcileExists(ctx context.Context, name types.NamespacedN
 	}
 	log.WithValues("objects", fmt.Sprintf("%d", len(objects.Items))).Info("built deployment objects")
 
+	if r.options.status != nil {
+		bool, err := r.options.status.VersionCheck(ctx, instance, objects)
```

Comment:
Nit: Don't name it `bool` - it's confusing.  e.g. maybe something like `validVersion, err := r.options.status.VersionCheck(...)`